src/screens/Dashboard/TripSearch.tsx

```diff
@@ -213,7 +218,20 @@ const TripSearch: React.FC<RootProps> = ({navigation}) => {
         rightButton={{type: 'chat'}}
         leftButton={{
           type: 'back',
-          onPress: () => navigation.goBack(),
+          onPress: () => {
+            if (callerRoute?.name) {
+              navigation.setParams({
+                callerRoute: undefined,
+              });
+
+              return navigation.navigate({
+                name: callerRoute?.name as any,
```

Comment:
Yes, that is what I intended but, the `ts` check doesn't like it without any, I think because it should navigate to one specific screen but callRoute.name could be any of them.

Reply:
Does `callerRoute?.name as TripSearchRouteName` work?

Reply:
Using `any` is in line with how it is done other places, like in `LocationSearch`. I think casting to `TripSearchRouteName` is not much better, as it is not actually a `TripSearchRouteName`.

Reply:
Yes, it works, but it is a bit strange. But should I use then `TripSearchRouteName` instead?